src/analysis/modals/CloudEnvironmentModal.ts

```diff
@@ -372,6 +372,9 @@ export const CloudEnvironmentModal = ({
     );
 
   const isCloudEnvModalDisabled = (toolLabel: ToolLabel): boolean => {
+    if (workspace.workspace.state === 'Deleting' || workspace.workspace.state === 'DeleteFailed') {
```

Comment:
Could this possibly be using cached state?

Reply:
I think you will need to call the `refreshWorkspace` callback from `useWorkspace` to force an update in the cache.

Reply:
The delete modal dumps you out on the list page anyway, so I just invalidated the cache after deleting.